src/targets/__tests__/symbolCollector.test.ts

```diff
@@ -0,0 +1,146 @@
+  bundleIdPrefix: 'bundleIdPrefix-',
+};
+
+function getSymbolCollectorInstance(
+  customConfig?: Record<string, unknown>
+): SymbolCollector {
+  const config = customConfig
+    ? customConfig
+    : {
+        ['testKey']: 'testVal',
+      };
+  return new SymbolCollector(
+    {
+      name: 'aws-lambda-layer',
+      ...config,
+    },
+    new NoneArtifactProvider()
+  );
+}
+
+describe('target config', () => {
+  test('symbol collector not present in path', () => {
+    (checkExecutableIsPresent as jest.MockedFunction<
+      typeof checkExecutableIsPresent
+    >).mockImplementationOnce(() => {
+      throw new Error('Checked for executable');
+    });
+
+    expect(getSymbolCollectorInstance).toThrowErrorMatchingInlineSnapshot(
+      `"Checked for executable"`
+    );
+    expect(checkExecutableIsPresent).toHaveBeenCalledTimes(1);
+    expect(checkExecutableIsPresent).toHaveBeenCalledWith(
+      SYM_COLLECTOR_BIN_NAME
+    );
+  });
+
+  test('config missing', () => {
+    (checkExecutableIsPresent as jest.MockedFunction<
+      typeof checkExecutableIsPresent
+    >) = jest.fn();
+
+    expect(getSymbolCollectorInstance).toThrowErrorMatchingInlineSnapshot(
+      `"Required configuration not found in configuration file. See the documentation for more details."`
+    );
+    expect(checkExecutableIsPresent).toHaveBeenCalledTimes(1);
+    expect(checkExecutableIsPresent).toHaveBeenCalledWith(
+      SYM_COLLECTOR_BIN_NAME
+    );
+  });
+
+  test('symbol collector present and config ok', () => {
+    (checkExecutableIsPresent as jest.MockedFunction<
+      typeof checkExecutableIsPresent
+    >) = jest.fn();
+
+    const symCollector = getSymbolCollectorInstance(customConfig);
+    const actualConfig = symCollector.symbolCollectorConfig;
+    expect(checkExecutableIsPresent).toHaveBeenCalledTimes(1);
+    expect(checkExecutableIsPresent).toHaveBeenLastCalledWith(
+      SYM_COLLECTOR_BIN_NAME
+    );
+    expect(actualConfig).toHaveProperty('serverEndpoint');
+    expect(actualConfig).toHaveProperty('batchType');
+    expect(actualConfig).toHaveProperty('bundleIdPrefix');
+  });
+});
+
+describe('publish', () => {
+  test('no artifacts found', () => {
+    const symCollector = getSymbolCollectorInstance(customConfig);
+    symCollector.getArtifactsForRevision = jest
+      .fn()
+      .mockReturnValueOnce(() => []);
+    expect(spawnProcess).not.toHaveBeenCalled();
+  });
+
+  test('with artifacts', async () => {
+    (withTempDir as jest.MockedFunction<typeof withTempDir>).mockImplementation(
+      async cb => await cb('tmpDir')
+    );
+    (spawnProcess as jest.MockedFunction<
+      typeof spawnProcess
+    >).mockImplementation(() => Promise.resolve(undefined));
+
+    const mockedArtifacts = [
+      { filename: 'artifact1', storedFile: { lastUpdated: 'tomorrow' } },
+      { filename: 'artifact2', storedFile: { lastUpdated: 'next week' } },
+      { filename: 'artifact2', storedFile: { lastUpdated: 'in 5 years' } },
```

Comment:
Are you sure these `lastUpdated` strings are valid? I think they should be ISO-8601 date strings?